### ISSUE TITLE
***Please Help***  Recovery scenario failing

### ISSUE THREAD

**Issue by prashant-dutt-mishra** [quickfix/quickfix#102]
I have added a unit test case in SessionTestCases.cpp which is failing.

Scenario:
1. Current Expected Target Sequence number is 10. and we send a login request.
2. Login response with sequence number 100 comes in and just after that Application Message with sequence number 101 comes.
3. Quickfix queues these message and sends resend request with 7=11 and 16=0.
4. Sequence reset with sequence number 11 and NewSeqNum=101 comes.
5. Quickfix thinks that the resend range has been satisfied.
6. Quickfix sets its ExpectedTargetSeqNum to 101. and starts processing the queued message. 
7. It first processes the Logon and increments its ExpectedTargetSeqNum=102
8. It then starts processing the queued Application Message with seq num 101.
9. Quickfix detects the seqnum is lower on that message than what is expected (102).
10. Sends a rejects and disconnect.

Has anyone came across this scenario before?
Is this Test case valid?

**Comment by anupanup2001** [comment]
After point 3, I believe in your scenario, there isn't any valid message between seq 10 - 100. Only heartbeat message I guess. Is that right?

This case seems valid to me. If this is the behavior, it might be a bug. Maybe some expert can have a look.

**Comment by prashant-dutt-mishra** [comment]
Yes Anup you are right.

After point 3. there are only Application Message on the server side so it will only replay message with seq num 101.

**Comment by dxwu88** [comment]
Which exchange are you connecting to? I don't have any problem for CME Globex.

**Comment by prashant-dutt-mishra** [comment]
This is not specific to CME's flow.

But yes I found this when I was adding test cases for the changes I was doing for CME.

**Comment by quickfix** [comment]
Is this a test case that came packaged with QuickFix, or one you added?

**Comment by prashant-dutt-mishra** [comment]
I added it.

But I think its not an issue. I added LoggerFactory in the UnitTest class and got the issue.

So after step 6, it will not process LogonMessage.
Reason: the current ExpectedTargetSeqNum is 101 and Logon has 34=100. So Logon will not be fetched and directly Application Message (seq num 101) will be processed. And it should be fine.

**Question: What will happen to the LogonMessage? Will it remain in the queue? Shouldn't we clear the Queue?**
